### ISSUE TITLE
Android 8.1 random connection issues with apps

### ISSUE THREAD

**Issue by johnwilkins128** [ukanth/afwall#914]
Hello,

This problem has only started on Lineage 15.1 (Android 8.1). Basically, some apps (all in permitted list) have connections issues: either they do not work at all or half-work. For example, messengers can receive messages but cannot send, market does not download apps. There is also a cross on wi-fi icon suggesting there is no Internet but funnily enough other apps work well.

If I reboot the phone, all works good for a few minutes and then the glitch starts again. If I disable afwall - cross disappear and everything works fine.

Sometimes the phone completely ignores wi-fi even though it is connected to it and all the traffic goes through sim card.

I tried to play with settings, restarted the phone many times, tried different versions (starting from 2.9.6 up until 3.0.1) - nothing helps. Occasionally I see an error in logcat:

command 'iptables -P OUTPUT DROP' exited with status 4, retrying (attempt 1/10) 

Sometimes the command is different but I don't understand where it comes from, there is no -P OUTPUT DROP anywhere.

I have attached IPv4rules.log and hope this could be resolved.

[IPv4rules.log](https://github.com/ukanth/afwall/files/2660111/IPv4rules.log)

**Comment by ukanth** [comment]
For downloads, you need to allow "Media server" and "Downloads Manager" . Enable also "Core" apps except "Any app"

Make sure you  "enable ipv6" from rules and preferences.

**Comment by johnwilkins128** [comment]
thanks!

Both "Media server" and "Downloads Manager" were enabled.

I enabled ipv6 - no effect.

Enabled core apps except any app - it works until I switch from wi-fi to mobile data, then stops working. After rebooting the phone everything works again until switching.

**Comment by EukalyptusX** [comment]
Yes, exact problem here. Same behavior, e.g. Signal Messenger can't send messages after some short time of inactivity of my phone. Receiving messages just works fine. Also other apps have problems with downloading after inactivity. After reboot it works for some time.

If you need any logs or something, please let me know.

I'm on latest build LineageOS 15.1 without GApps.

**Comment by anmith** [comment]
Same problem here. I'm on latest LineageOS 15.1 without GApps and have some whitelisted apps with connection problems.
ipv6 is enabled, core apps are also enabled.
One of the affected apps is F-Droid which is only able to get updates when I disable AFWall and reboot the device afterwards.

**Comment by EukalyptusX** [comment]
Still have this issue in latest LineageOS 16.0
I think I know the cause: Activating support for IPv6 is responsible for the issue.

If you deactivate it, everything seems to be fine. Activating it provokes the issue.

**Comment by Tobias-B-Besemer** [comment]
I had some issues with apps not able to get i-net connection (e.g. "XDA Lab" [Store])... I was able to fix them with deactivating the AFWall+ Module in Xposed. I have IPv6 support, but IPv6 is blocked that no traffic can bypass my IPv4 VPN...
I still have the issue that my phone sometimes loss contact to WiFi and isn't able after that to find any WiFi again till I reboot... But I'm not sure if this is an issue from AFWall+...

**Comment by EukalyptusX** [comment]
Earlier I thought that as well, so I completely wiped my device, and flashed everything, so I had a clean LineageOS without custom kernel and without GApps. Just rooted my device (SuperSU or Magisk, tried both) and installed Afwall. I didn't install Xposed (Xposed isn't available on Android 9 anyway).
The problem still existed on that clean ROM. Deactivating IPv6 was the only option that helped for me. Sadly. :-(

**Comment by PixsaOJ** [comment]
I don't know if its the same issue since im on android 15. But i think for me disabling incomming connections setting worked.
